dnfdaemon-client/commands/repolist/repolist.cpp

```diff
@@ -102,35 +102,25 @@ class RepoDbus {
     dnfdaemon::KeyValueMap rawdata;
 };
 
-// repository list table columns
-enum { COL_REPO_ID, COL_REPO_NAME, COL_REPO_STATUS };
-
-static struct libscols_table * create_repolist_table(bool with_status) {
-    struct libscols_table * table = scols_new_table();
-    if (isatty(1)) {
-        scols_table_enable_colors(table, 1);
-        scols_table_enable_maxout(table, 1);
-    }
-    struct libscols_column * cl = scols_table_new_column(table, "repo id", 0.4, 0);
-    scols_column_set_cmpfunc(cl, scols_cmpstr_cells, NULL);
-    scols_table_new_column(table, "repo name", 0.5, SCOLS_FL_TRUNC);
-    if (with_status) {
-        scols_table_new_column(table, "status", 0.1, SCOLS_FL_RIGHT);
+class QueryRepoDbus {
+public:
+    QueryRepoDbus(dnfdaemon::KeyValueMapList & repositories) {
+        for (auto raw_repo : repositories) {
+            queryrepo.push_back(new RepoDbus(raw_repo));
+        }
     }
-    return table;
-}
 
-static void add_line_into_table(
-    struct libscols_table * table, bool with_status, const char * id, const char * descr, bool enabled) {
-    struct libscols_line * ln = scols_table_new_line(table, NULL);
-    scols_line_set_data(ln, COL_REPO_ID, id);
-    scols_line_set_data(ln, COL_REPO_NAME, descr);
-    if (with_status) {
-        scols_line_set_data(ln, COL_REPO_STATUS, enabled ? "enabled" : "disabled");
-        struct libscols_cell * cl = scols_line_get_cell(ln, COL_REPO_STATUS);
-        scols_cell_set_color(cl, enabled ? "green" : "red");
+    ~QueryRepoDbus() {
+        for (auto repo : queryrepo) {
+            delete [] repo;
```

Comment:
`delete []` ends with a segfault.
Plain `delete` works fine.

Reply:
Please don't do this at all :slightly_smiling_face: use `std::unique_ptr` instead of raw pointers.